### PR TITLE
JS: Fix end-of-line comments if followed by trailing whitespace

### LINKED ISSUE #8068: End-of-line comments aren't formatted properly if followed by trailing whitespace
**Prettier 2.0.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA3AhgJwATuwXm2GwHoAqbOADxkzwHcALAS3gGcAHdMObAWj4A+bGRLZsAHSjjxAI2wBfANxSpGHHkLFy2KBGxNWcTt14DhoqTOzzlIADQgIHGM2htkoLJgj0AClgIHijoADb06ACeHo6ydGAA1nAwAMpcYMxQAObItACucI6MMAC2oQDqLOzpcClBrMyorJHI4GwxIJlscJgwfnRZJejIAGZh3Y4AVmxUAELxSanoJXAAMplwo+OFINNUKZlZoXAAinkQ8FuhEyBcmN2YrbLosnChDreYmTDlzAAmMEYyAAHAAGRwcHzdcp0DitSHGHqoTaOACO53g-WcwRA6DYfCgcDgf2JH0wcHRzHJ-XQg2GSDG1x23RKzFymAKjjYh2OZwumwZ20cMBevwBQKQACZhXRmKFDgBhCAlIatYwAVg+eW6ABUXsFGTdUAUAJJQEmwFJgL4uACC5pSMEixyu3QUCiAA)
```sh
--parser babel
```

**Input:**
```jsx
var a = { /* extra whitespace --> */  
    b };

var a = { /* no whitespace --> */
    b };
```

**Output:**
```jsx
var a = {
  /* extra whitespace --> */

  b,
};

var a = {
  /* no whitespace --> */ b,
};

```

**Expected behavior:**
```jsx
var a = {
  /* extra whitespace --> */ b,
};

var a = {
  /* no whitespace --> */ b,
};

```

### PR DESCRIPTION
**Pull request by shisama**
If comments line has trailing whitespaces, comments aren't detected as end-of-line.

Fix #8068

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/main/comments.js [hunk 1]
```diff
@@ -521,7 +522,11 @@ function printComments(path, print, options, needsSemi) {
 
       const text = options.originalText;
       const index = skipNewline(text, options.locEnd(comment));
```
Comment (thorn0) [inline src/main/comments.js:524]:
```suggestion
      const index = skipNewline(text, skipSpaces(text, options.locEnd(comment)));
```

File: src/main/comments.js [hunk 2]
```diff
@@ -521,7 +522,11 @@ function printComments(path, print, options, needsSemi) {
...
-      if (index !== false && hasNewline(text, index)) {
+      if (
+        index !== false &&
+        hasNewline(text, index) &&
+        !hasSpaces(text, index)
```
Comment (thorn0) [inline src/main/comments.js:528]: As the tests show, this is not a correct solution.

**Comment by shisama** [comment]
@thorn0 I fixed them. Thank you.

File: tests/comments/trailing_space.js [hunk 3]
```diff
@@ -3,3 +3,8 @@
 // Do not trim trailing whitespace from this source file!
 
 // There is some space here ->                        
+
+var a = { /* extra whitespace --> */  
+  b };
+var a = { /* no whitespace --> */
+  b };
```
Comment (thorn0) [inline tests/comments/trailing_space.js:10]: Please create a new test file. Don't modify existing ones.

Comment (shisama) [reply tests/comments/trailing_space.js]: Thanks. I created a new test file. Please check it.

**Comment by fisker** [comment]
Can we use [snippets](https://github.com/prettier/prettier/pull/8088) to test the whitespace? It will never get lost. You might need rebase first.

**Comment by fisker** [comment]
Thank you for contributing to Prettier!

**Pull request merged**